RecordHub.CatalogService/RecordHub.CatalogService.Infrastructure/Services/CatalogCheckerService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Grpc.Core;
```

Comment:
I would recommend you to locate this in API project. Anyway it is your API just implemented with another technology. 
But if you will (you will :) ) create client for this grpc server, then it will be located inside infrastructure project.

Reply:
I had problems with endpoints of grpc and usual api, because grpc working on http 2, and i didn't manage to make it work in one project(I tried to set them on different ports in appsetings.json, but its not worked for me) so i put it in another project catalogservice.grpc. Client of my grpc service is basket service, it asking for availability when adding to cart.

Reply:
So does your gRPC proj act like a partial for API? Actually not, but if i understand correctly, it's seems to be.